PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/ConnectionStringCheck.java

```diff
@@ -0,0 +1,44 @@
+package com.microsoft.azure.toolkit.intellij.azure.sdk.buildtool;
+
+import com.intellij.codeInspection.*;
+import com.intellij.psi.*;
+import com.microsoft.applicationinsights.TelemetryClient;
+import org.jetbrains.annotations.NotNull;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This class extends the LocalInspectionTool and checks for the usage of connection strings in the code.
+ * Connection strings that start with "DefaultEndpointsProtocol=http;AccountName=" are flagged as problems.
+ * This is because the use of such connection strings is not recommended.
+ */
+public class ConnectionStringCheck extends LocalInspectionTool {
+
+    @NotNull
+    @Override
+    public PsiElementVisitor buildVisitor(@NotNull ProblemsHolder holder, boolean isOnTheFly) {
+
+        TelemetryClient telemetry = TelemetryClientProvider.getTelemetryClient();
+
+        return new JavaElementVisitor() {
```

Comment:
I don't believe we need to use anonymous classes. It makes it harder to read/test. Consider using a lambda or a named class.

Reply:
Using a named class. Also makes testing easier. https://gist.github.com/conniey/5e188d78102a789ef3901cb50e014039

Reply:
yehp I agree - I've been using named classes for subsequent rules and it is much easier thank you:)

Reply:
No reason to not start doing it from this PR onwards, and followed subsequently.

Reply:
configured to add named class:)